### ISSUE TITLE
Output is not set when 'push' is set to false

### ISSUE THREAD

**Issue by rkeidar** [docker/build-push-action#387]
### Troubleshooting

Before sumbitting a bug report please read the [Troubleshooting doc](https://github.com/docker/build-push-action/blob/master/TROUBLESHOOTING.md).

#### Steps to reproduce this issue

1.  Set 'push: false'
2.  Create another action which use the output '${{ steps.docker_build.outputs.digest }} "
3. The output is not set although the image is being created. 

#### Expected behaviour

The outputs.digest should be set so I can use it on other actions. 

#### Actual behaviour

Output is enpty

### Configuration

* Repository URL (if public): 
* Build URL (if public): 

```yml
 #--------------  Build  Dockerfile ------------
 
      - name: Build  Docker image
        id: docker_build
        uses: docker/build-push-action@v2.5.0
        with:
          file: ${{ env.DOCKER_FILE }}
          push: true
          tags:  ${{secrets.CONTAINER_REGISTRY}}/${{ env.BASE_CONTAINER_TAG }}:${{ env.APP_VERSION }}-${{ env.STATE_TAG }}
 #--------------  Push to registry ------------
 
      - name: Push Docker image
        run: |
          echo "steps.build.outputs.digest  = ${{ steps.docker_build.outputs.digest }} "
```

### Logs

> Download the [log file of your build](https://docs.github.com/en/actions/managing-workflow-runs/using-workflow-run-logs#downloading-logs) and [attach it](https://docs.github.com/en/github/managing-your-work-on-github/file-attachments-on-issues-and-pull-requests) to this issue.

**Comment by crazy-max** [comment]
@rkeidar Works fine on my side with the following workflow:

```Dockerfile
FROM alpine
RUN echo "Hello world!"
```

```yaml
name: digest

on:
  push:

jobs:
  digest:
    runs-on: ubuntu-latest
    steps:
      -
        name: Checkout
        uses: actions/checkout@v2
      -
        name: Build
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          context: .
      -
        name: Digest
        run: |
          echo ${{ steps.docker_build.outputs.digest }}
```

Can you give me the link to your repo?

**Comment by rkeidar** [comment]
Thanks for looking at this issue.
 I've tested your action and it seems to work fine on my side also. 
After debugging I've firured out that we are also using : 
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1

That cause this issue. It seems that we don't really need it in the workflow. When I removed it ,everything works fine.

Thanks a lot for looking into it . It can be probably be closed.
